### PR TITLE
[61] [Test] Side Menu screen

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/flutter-survey/issues/61

## What happened

Write tests for Side Menu screen
 
## Proof Of Work

Screen Shot 2021-05-18 at 09 42 40

### CODE REVIEW

File: test/modules/home/home_presenter_test.dart [hunk 1]
```diff
@@ -171,5 +171,35 @@ void main() {
         verify(view.showUser(user)).called(1);
       });
     });
+
+    describe("it's sideMenuDidShow emits", () {
```
Comment (manh-t) [inline test/modules/home/home_presenter_test.dart:175]:
```suggestion
    describe("its sideMenuDidShow emits", () {
```
And same for the rest

**Review by manh-t — COMMENTED** [review]
just grammar errors, rest sgtm

File: test/modules/home/home_view_test.dart [hunk 2]
```diff
@@ -238,5 +244,54 @@ void main() {
         expect(delegate.showDetailButtonDidTap, emits(surveys.first));
       });
     });
+
+    describe("it's showSideMenu() is called", () {
```
Comment (sleepylee) [inline test/modules/home/home_view_test.dart:248]: - sometimes you have the () sometimes you don't, let's be consistent.
- `its` not `it's` here I believe

File: test/modules/home/home_view_test.dart [hunk 3]
```diff
@@ -238,5 +244,54 @@ void main() {
...
+      beforeEach((tester) async {
+        await tester.pumpAndSettle();
+        module.view.showSideMenu();
+        await tester.pumpAndSettle();
+      });
+
+      it("triggers delegate's sideMenuDidShow emits", (tester) async {
```
Comment (sleepylee) [inline test/modules/home/home_view_test.dart:255]: triggers delegate sideMenuDidShow to emit - remember the last conversation we discussed

File: test/modules/home/home_view_test.dart [hunk 4]
```diff
@@ -238,5 +244,54 @@ void main() {
...
+          await tester.pumpAndSettle();
+        });
+
+        it("triggers delegate's sideMenuDidDismiss emits", (tester) async {
+          expect(delegate.sideMenuDidDismiss, emits(null));
+        });
+      });
+    });
+
+    describe("it's user avatar button is tapped", () {
```
Comment (sleepylee) [inline test/modules/home/home_view_test.dart:272]: same here, the grammar structure is not very consistent you might need to recheck them all 🙏

**Review by manh-t — APPROVED** [review]

**Review by sleepylee — APPROVED** [review]

**Pull request merged**